webpack/components/extensions/HostDetails/Cards/ContentViewDetailsCard.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+  Card,
+  CardHeader,
+  CardTitle,
+  CardBody,
+  Flex,
+  FlexItem,
+  GridItem,
+  Label,
+} from '@patternfly/react-core';
+
+import { urlBuilder } from 'foremanReact/common/urlHelpers';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { propsToCamelCase } from 'foremanReact/common/helpers';
+import PropTypes from 'prop-types';
+import ContentViewIcon from '../../../../scenes/ContentViews/components/ContentViewIcon';
+
+const renderContentViewDetails = (hostDetails) => {
+  const hostDetailsData = propsToCamelCase(hostDetails.content_facet_attributes);
+  const {
+    contentView,
+    lifecycleEnvironment,
+    contentViewVersionId,
+    contentViewVersion,
```

Comment:
Now that we're refactoring this to be a component, I think it would be good to add a `propTypes` object for these as well.  Could be a shape, like
```js
hostDetails: PropTypes.shape(
{
  content_view: PropTypes.shape({}),
  lifecycle_environment: PropTypes.shape({}),
  content_view_version_id: PropTypes.string,
...
}
)
```

Another option is to move everything up one level and have those items be props (not under the `hostDetails` object). Then you'd have to use `propsToCamelCase` with props spread syntax in the parent component, like
```js
<HostContentViewDetails {...propsToCamelCase(hostDetails.content_facet_attributes)} />
```
Completely up to you which makes more sense to you.